### PR TITLE
[IE Samples] Activating new parameter is compact mode(memory_reuse) in speech sample

### PR DESCRIPTION
**Pull request by mgrdv**
Details:
 - [IE Samples] Activating new parameter is compact mode in speech sample

Tickets:
 - 82623

### CODE REVIEW

File: samples/cpp/speech_sample/main.cpp [hunk 1]
```diff
@@ -243,7 +243,7 @@ int main(int argc, char* argv[]) {
         };
         gnaPluginConfig[ov::intel_gna::execution_target.name()] = parse_target(FLAGS_exec_target);
         gnaPluginConfig[ov::intel_gna::compile_target.name()] = parse_target(FLAGS_compile_target);
-        gnaPluginConfig[ov::intel_gna::memory_reuse.name()] = false;
+        gnaPluginConfig[ov::intel_gna::memory_reuse.name()] = !FLAGS_disable_compact_mode;
```
Comment (mryzhov) [inline samples/cpp/speech_sample/main.cpp:246]: I would suggest using the new name of this option - memory_reuse, with possible values (on/off or true/false)

Comment (mgrdv) [reply samples/cpp/speech_sample/main.cpp]: renamed the option

File: samples/cpp/speech_sample/speech_sample.hpp [hunk 2]
```diff
@@ -51,6 +51,9 @@ static const char enable_log_message[] = "Optional. Enable GNA logging, which ma
 /// @brief message for performance counters
 static const char performance_counter_message[] = "Optional. Enables per-layer performance report.";
 
+/// @brief message for disabling of compact (memory_reuse) mode
+static const char memory_reuse_message[] = "Optional. Enables compact mode (memory_reuse).";
```
Comment (mryzhov) [inline samples/cpp/speech_sample/speech_sample.hpp:55]:
```suggestion
static const char memory_reuse_message[] = "Optional. Disables memory optimizations for compiled model.";
```

File: samples/cpp/speech_sample/speech_sample.hpp [hunk 3]
```diff
@@ -130,6 +133,9 @@ static const char pwl_max_error_percent_message[] = "Optional. The maximum perce
 /// \brief Define flag for showing help message <br>
 DEFINE_bool(h, false, help_message);
 
+/// \brief Define flag for disabling compact (memory_reuse) mode <br>
+DEFINE_bool(memory_reuse, false, memory_reuse_message);
```
Comment (mryzhov) [inline samples/cpp/speech_sample/speech_sample.hpp:137]: memory_reuse_off ?

Comment (mgrdv) [reply samples/cpp/speech_sample/speech_sample.hpp]: ok

**Review by mryzhov — APPROVED** [review]

**Review by vladimir-dudnik — APPROVED** [review]

**Pull request merged**